### DISCUSSION TITLE
Insert interface HA in HACS

### DISCUSSION THREAD

**Discussion by djgel** [Yogui79/IntexPureSpa#11] [Ideas]
Add um HACS your interface Home Assistant  Intexspa, this is make more easy for all install this!!! 

https://hacs.xyz/

Tks!

**Comment by LEECHER1** [comment]
Hey,

Cool it is possible to Export/Import a complete interface?

**Comment by djgel** [comment]
Edit lovelace and copy you card ... i study for  import this in hacs .... 

My Cards :

- icon: mdi:hot-tub
    title: PureSpa
    path: purespa
    badges: []
    cards:
      - type: grid
        cards:
          - type: gauge
            entity: sensor.actual_temperature
            min: 0
            max: 100
          - type: gauge
            entity: sensor.set_temp
            min: 0
            max: 100
          - type: entity
            entity: sensor.status
        columns: 3
        square: false
      - type: button
        tap_action:
          action: toggle
        entity: switch.change_farenheit_celsius
        show_state: true
        icon: mdi:power
        show_name: false
        icon_height: 95px
        hold_action:
          action: toggle
      - type: conditional
        conditions:
          - entity: switch.intex_on
            state: 'on'
        card:
          type: horizontal-stack
          cards:
            - type: thermostat
              entity: climate.intex_actual_temperature
              name: Ajuste de  Temperatura PureSpa
      - type: conditional
        conditions:
          - entity: switch.intex_on
            state: 'on'
        card:
          type: entities
          entities:
            - entity: switch.intex_bubble
              icon: mdi:chart-bubble
              name: Bolhas
            - entity: switch.intex_jets
              icon: mdi:waves
              name: Hidromassagem
            - entity: switch.intex_heat
              name: Aquecimento
              icon: mdi:radiator
            - entity: switch.intex_filter
              name: Filtragem
              icon: mdi:air-filter
            - entity: switch.intex_sanitizer
              name: Sanitização
              icon: mdi:shield-sun
      - type: conditional
        conditions:
          - entity: switch.intex_on
            state: 'on'
        card:
          type: entity
          entity: sensor.brasilia_temperature
          name: Temp. em Brasilia
          unit: C

**Comment by BenoitB31** [comment]
Bonjour, je ne sais pas si quelqu'un parle français mais j'aurais besoin d'aide je n'arrive pas a faire 'interface. La communication et le MQTT ok.

Merci

**Reply by Yogui79** [reply]
Bonjour @BenoitB31 

quel interface n'arrive tu pas faire ?

**Reply by BenoitB31** [reply]
Je n'arrive pas a faire mon Lovelace, j'ai essayé de prendre exemple sur celui proposer plus haut mais rien de marche. 
Si quelqu'un peut me partager le sien ou celui de la video de l'exemple.
Merci de votre aide. Home Assistant 2021.12.10

**Reply by Yogui79** [reply]
@BenoitB31 
dsl j'ai pas HA. 
@LEECHER1  l'utilise mais il parle pas francais que Anglais ou Allemand

**Reply by rommess** [reply]
@BenoitB31 tu as du avancer depuis le temps, mais si jamais, regarde mon commentaire sur cette page ;)
https://github.com/Yogui79/IntexPureSpa/issues/44

**Comment by LEECHER1** [comment]
Hey,

Je traduis cela avec Google Translate.
Veuillez vérifier les chaînes MQTT avec le courtier MQTT (Mosquitto) pour voir si elles fonctionnent.
Vous pouvez ensuite facilement le reconstruire et le contrôler dans Lovelace.

Je te donnerai mes dates  aujourd'hui ou demain soir.  Certaines de mes données sont en allemand, veuillez le noter.

Je vous remercie
LEECHER1

**Reply by BenoitB31** [reply]
Merci beaucoup, mes chaines MQTT sont fonctionnelles. Mais c'est dans le Lovelace que je n'arrive pas a créer mes actions.

**Comment by LEECHER1** [comment]
Salut,

Je vous donne mes données de :
- "Lovelace"
- "Configuration.yaml" (données Intex uniquement)
- "scripts.yaml"
- "automatisations.yaml"

Tous dépendent les uns des autres

Veuillez noter que de nombreux noms de capteurs et autres textes en allemand devraient être remplacés.

Il est préférable que vous le créiez vous-même et que vous n'utilisiez mes données que comme modèle.

[https://we.tl/t-4VEw93Wk0s](url)

j'espère que j'ai pu t'aider

Meilleures salutations
LEECHER1

PS : tout ne fonctionne que s'il y a une connexion avec la pompe.

**Reply by BenoitB31** [reply]
Merci

**Reply by Skyber1967** [reply]
Hi LEECHER1,

Your link with your configuration seems no more valid. I am very interested to see how you did :)
Can you share it again please ?

Best regards

**Reply by madhehler** [reply]
Hi LEECHER1

I really would need some help with the integration in HOMEassist.
might you pleeeeease help me

Greetings
Mat

**Comment by madhehler** [comment]
hello 

but which IP adress I should take, the one from the EPS32?

Greetinsg
Mat

**Reply by madhehler** [reply]
I really struggle to integrate the ESP into Homeassist.
I think the board is not even connected 
How can I conenct this?

**Reply by BenoitB31** [reply]
You put a free IP of your home network on the esp32. At first you can Ping this IP to Control if it’s Connected. Sorry for my english 🥲

**Reply by madhehler** [reply]
Ping is ok. And available at the Fritzbox. 
But how to import the esp into HA. 
(Deutsch geht auch)

**Comment by LEECHER1** [comment]
Hey Madheler,

Ich versuche es mal in deutsch. Ist einfacher für mich, wenn es okay ist.
Kannst du mir nochmal genau dein problem schreiben?

Hier ein sehr grober Aufbau wie es funktioniert:

Im ersten Schritt ist es wichtig, dass du eine fixe IP zum ESP32 hast.
Anschließend musst du den MQTT broker installieren im HomeAssistant. (
https://smart-home-assistant.de/mqtt-server-installieren/)
und dann sollte die Verbindung zum ESP32 bestehen.
[image: image.png]

weiter geht es dann mit den configuration.yaml hier die verlinkung: (Bitte
nur als beispiel sehen nu musst es an deine umgebung anpassen)
https://github.com/Yogui79/IntexPureSpa/blob/main/src/HomeAutomation/HomeAssistant/configuration.yaml

Da musst du die MQTT topic & payload eintragen und anschließend in der
benutzeroberfläche die Sensoren einrichten.

lg LEECHER1



Am Fr., 28. Apr. 2023 um 13:49 Uhr schrieb madhehler <
***@***.***>:

**Comment by madhehler** [comment]
Mann super das du mir hilft, kannst auch mal direkt schreiben an
***@***.***

Also ich bin eigentlich ganz gut im Netzwerksachen aber programmieren ist
nicht sooo meins.

Also die Verbindung mit der Pumpe geht - über Putty bekomm ich die
Verbindung und z.b Temperatur
MQTT geht, soweit, da ich meine Solar Anlage erfolgreich eingebunden habe

Aber ich habe irgendwie das Gefühl, dass das ESP noch nicht im HA drin ist.

Ich hab keine Verbinung und MRQT Broker ist installiert
Was muss ich denn anpassen in der YAML?

Gruss
Hehler

[image: image.png]

Meinst

[image: image.png]



Am Fr., 28. Apr. 2023 um 16:10 Uhr schrieb LEECHER1 <
***@***.***>:

**Reply by LEECHER1** [reply]
Hallo,

leider funktioniert des mit der E-Mail Adresse bei GitHub nicht. 

Du musst für die Konfiguration vom dem MQTT Broker in die Einstellungen > Geräte & Dienste > dann müsste einen Eintrag zu finden sein mit Mosquitto Broker geben > Konfigurieren > MQTT Einstellungen > hier musst die IP, Port, Benutzername & Passwort eintragen vom ESP32 (Was du im Spa.ino eingetragen hast)

Dann sollte es eine Verbindung geben. 

![Screenshot_2023-04-28-18-34-09-687_io homeassistant companion android~2](https://user-images.githubusercontent.com/12032696/235212701-8386f303-e216-4556-a178-ee7fe9126537.jpg)

Probiere das mal und gebe bescheid.

Lg

**Comment by madhehler** [comment]
Hi

das hab ich auch schon gemacht, sehe aber keine Entities nur no
communication

Ich hab das Gefühl, dass keine Verbindung mit dem ESP besteht. In der
Fritzbox ist das ESP da, aber mit einer anderen IP, was ja richtig ist.

Gibt es irgendwelche Test im Termin, damit ich die MQTT Verbindung testen
kann?
Oder gibt es eine Möglichkeit sich auf das ESP zu verbinden - Putty bekomm
ich nicht hin über die Fritzbox

Gruss
Hehler


[image: image.png]


[image: image.png]


Am Fr., 28. Apr. 2023 um 19:22 Uhr schrieb LEECHER1 <
***@***.***>:

**Reply by LEECHER1** [reply]
du kannst im gleich Feld im HomeAssistant auch die Toppics und Payloads testen ob was rein kommt:
<img width="560" alt="image" src="https://user-images.githubusercontent.com/12032696/235321530-f2c612ca-1f16-4ca9-b7e8-ce41d1b53348.png">

Hast du unter Addons > Mosquitto broker > Konfiguration deine Login Daten eingeben und das mit dem Certificate gemacht ? Sie auf der gleichen Seite "Dokumentation" (den genaueren weg weiß ich selber nicht mehr)

Ist der Port richtig im EPS32 und im HA angeben? 
Blockt evtl. dein Router den Port? 

Teste das mal.

lg LEECHER1

**Comment by madhehler** [comment]
Also im MQTT explorer sehe ich die temp , aber nur die temp?

Ich glaub die yaml ist einfach nicht correct. Ich habe gesehen das Pattern
nicht mehr unterstützt wird - ich hab die aus dem "Issues" im blog
genommen, aber keine Ahnung

Gibt es nicht einfach ein YAML die mal 1 zu 1 nehmen kann?

Gruss
Hehler

[image: image.png]

Am Sa., 29. Apr. 2023 um 21:54 Uhr schrieb LEECHER1 <
***@***.***>:

**Comment by madhehler** [comment]
finally it worked with the yaml file from  configuration.yaml does not
parse properly for HomeAssistant
<https://github.com/Yogui79/IntexPureSpa/issues/44>
The problem was that I already had my solar MQTT sensors which blocked the
Intex one
I just to find out how to use 2 yaml files and include them
Do you know how this works?
This option below with mqtt more is not working

but many thanks for your help
[image: image.png]

Am So., 30. Apr. 2023 um 13:38 Uhr schrieb Mat Heller ***@***.***>:
